### PR TITLE
refactor: Merge four test functions into a single test

### LINKED ISSUE #22: Refactor DeactivationNoticeTestCase

### PR DESCRIPTION
**Pull request by AdderAdder**
Fixes #22

### CODE REVIEW

File: zerver/tests/test_signup.py [hunk 1]
```diff
@@ -87,33 +87,32 @@ def test_cookie_data(self) -> None:
                                     'is_signup': True})
 
 class DeactivationNoticeTestCase(ZulipTestCase):
-    def test_redirection_for_deactivated_realm(self) -> None:
```
Comment (OscarMelin) [inline zerver/tests/test_signup.py]: Extra empty line

File: zerver/tests/test_signup.py [hunk 2]
```diff
@@ -87,33 +87,32 @@ def test_cookie_data(self) -> None:
...
-        realm = get_realm("zulip")
-        realm.deactivated = True
-        realm.save(update_fields=["deactivated"])
 
-        for url in ('/register/', '/login/'):
-            result = self.client_get(url)
-            self.assertEqual(result.status_code, 302)
-            self.assertIn('deactivated', result.url)
+    def test_deactivation_notice(self):
```
Comment (OscarMelin) [inline zerver/tests/test_signup.py]: Extra empty line

File: zerver/tests/test_signup.py [hunk 3]
```diff
@@ -87,33 +87,32 @@ def test_cookie_data(self) -> None:
...
 
-    def test_redirection_for_active_realm(self) -> None:
-        for url in ('/register/', '/login/'):
+        def test_redirect(url: str, expected_status_code: int, url_contains_string: str=None, body_contains_string: str=None):
```
Comment (OscarMelin) [inline zerver/tests/test_signup.py]: @saraervik Inner functions can be used in python when we want to hide them from a global scope, which is what we want do here 😄

File: zerver/tests/test_signup.py (hunk 2 above)
Comment (OscarMelin) [reply zerver/tests/test_signup.py]: Ignore this comment, empty line here aligns with overall coding style

**Comment by AdderAdder** [comment]
@OscarMelin I think I fixed the empty lines problem.

@saraervik If you still want the nested help function to be extracted just let me know. I am a little bit confused at the moment about whether we want it nested or not.

**Pull request merged**